src/gamestate/system_state.cpp

```diff
@@ -934,11 +934,17 @@ std::string_view state::to_string_view(dcon::unit_name_id tag) const {
 }
 
 dcon::trigger_key state::commit_trigger_data(std::vector<uint16_t> data) {
-	if(data.size() == 0) {
+	if(data.empty()) {
 		if(trigger_data_indices.empty())
 			trigger_data_indices.push_back(int32_t(0));
 		return dcon::trigger_key();
 	}
+	if(trigger_data_indices.empty()) { // Create placeholder for invalid triggers
+		trigger_data_indices.push_back(0);
+		trigger_data.push_back(uint16_t(trigger::integer_scope));
```

Comment:
Why are you using an integer scope, which requires an extra byte?

Reply:
A dummy trigger and effect would:

a) run on any scope (province, nation, state or pop)
b) evaluate to false, always

I found the integer_scope to fullfill those two criteria

Reply:
Then just use always = no. That would be a single uint16_t